rules/ast/ast.go

```diff
@@ -888,8 +888,12 @@ func (node *StringFunctionCall) Eval(timestamp clientmodel.Timestamp) string {
 // ----------------------------------------------------------------------------
 // Constructors.
 
-// NewScalarLiteral returns a ScalarLiteral with the given value.
-func NewScalarLiteral(value clientmodel.SampleValue) *ScalarLiteral {
+// NewScalarLiteral returns a ScalarLiteral with the given value. If sign is "-"
+// the value is negated.
+func NewScalarLiteral(value clientmodel.SampleValue, sign string) *ScalarLiteral {
+	if sign == "-" {
```

Comment:
From IRC:

13:22 <@jrv> fabxc: one thing about the new one: for consistency, I did all string parsing stuff in helpers.go
13:23 <@jrv> fabxc: that's why there were all the wrappers in there
13:23 <@jrv> fabxc: and then in "ast" I only pass in what's already "parsed" and verified
13:23 <@jrv> fabxc: this probably won't matter much if we replace the parser anyways, but for now, could you do the "-" parsing in helpers.go and just pass in the right value to the existing ast.NewScalarLiteral?

:+1: Otherwise!